test/serializers/associations_test.rb

```diff
@@ -290,6 +289,61 @@ def test_illegal_conditional_associations
           assert_match(/:if should be a Symbol, String or Proc/, exception.message)
         end
       end
+
+      class InheritedSerializerTest < ActiveSupport::TestCase
+        InheritedPostSerializer = Class.new(PostSerializer) do
```

Comment:
@nullvoxpopuli let's not introduce more dynamics classes right now unless we need to